client/modules/home.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react';
+import {
+  Card,
+  CardContent,
+  Typography,
+  Grid,
+  CircularProgress,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/styles';
+import { useSelector, useDispatch } from 'react-redux';
+import { SomeComponent } from 'client/components';
+import { AppStoreState } from 'client/store/reducers';
+import { chapterActions } from 'client/store/actions';
+
+const useStyles = makeStyles(() => ({
+  root: {
+    padding: 15,
+  },
+}));
+
+const renderCardContent = (loading, name, description) => {
```

Comment:
1. dont call component `render` + component name, this isn't a render prop
2. Maybe it would be better to move the logic up to the parent component
3. For something like this where you only have a return it would be better to use a shorthand ES6 syntax
```
const foo = (a: number, b: number) => a + b;
```

Reply:
Not sure how everyone else thinks about this, but I usually tend to separate big component hierarchy, into separate functions, just for readability's sake. This also helps in separating out JS loops and conditionally created components inside render.

This could've been called, `getCardContentWithLoader()` for better naming or even separated out to it's own smaller UI Component for re-usability. I think latter seems a better option as per ReactJS norms, where composition is preferred.

Learnt a new pattern, here. :sweat_smile:

Reply:
hmmm, I would definatly call it getCardContent or something like that, which implies that it returns JSX. Also this can be made into a shorter syntax. But I would separete the loading out into the parent component and use `?` insead of `variable &&` and `!variable &&`

Reply:
The `Component` looks much better now. :tada: